x/mongo/driver/dns/dns.go

```diff
@@ -113,15 +113,18 @@ func (r *Resolver) fetchSeedlistFromSRV(host string, srvName string, stopOnErr b
 func validateSRVResult(recordFromSRV, inputHostName string) error {
 	separatedInputDomain := strings.Split(strings.ToLower(inputHostName), ".")
 	separatedRecord := strings.Split(strings.ToLower(recordFromSRV), ".")
-	if len(separatedRecord) < 2 {
-		return errors.New("DNS name must contain at least 2 labels")
+	if l := len(separatedInputDomain); l < 3 && len(separatedRecord) <= l {
+		return fmt.Errorf("DNS name must contain at least %d labels", l+1)
```

Comment:
This error is sort of ambiguous to me. What are your thoughts on matching what node.js does? https://github.com/mongodb/node-mongodb-native/pull/4197/files#diff-39b2554fd18da165b59a6351b1aafff3714e2a80c1435f2de9706355b4d32351R1181

```go
fmt.Errorf(Server record (%d) does not have at least one more domain level than parent URI (%d), l, len(seperatedRecord))
```